src/client/pydaos/pydaos_shim.c

```diff
@@ -222,12 +222,23 @@ cont_open(int ret, char *pool, char *cont, int flags)
 	if (prop)
 		daos_prop_free(prop);
 	if (rc) {
-		if (daos_handle_is_valid(oh))
-			daos_kv_close(oh, NULL);
-		if (daos_handle_is_valid(coh))
-			daos_cont_close(coh, NULL);
-		if (daos_handle_is_valid(poh))
-			daos_pool_disconnect(poh, NULL);
+		int	rc2;
+
+		if (daos_handle_is_valid(oh)) {
+			rc2 = daos_kv_close(oh, NULL);
+			if (rc2)
+				D_ERROR("failed to close kv\n");
+		}
+		if (daos_handle_is_valid(coh)) {
+			rc2 = daos_cont_close(coh, NULL);
+			if (rc2)
+				D_ERROR("failed to close container\n");
+		}
+		if (daos_handle_is_valid(poh)) {
+			rc2 = daos_pool_disconnect(poh, NULL);
+			if (rc2)
+				D_ERROR("failed to disconnect pool\n");
+		}
```

Comment:
So these fixes are just about checking the return code, right? We don't need to `return rc2` if case `rc == 0 && rc2 != 0`?

Reply:
if rc == 0 we won't be here. and we don't want to overwrite rc with rc2.
so yes just to silence coverity

Reply:
derp. I should really read above before commenting :(